org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/validation/MarkerCreator.java

```diff
@@ -42,17 +42,16 @@ protected void setMarkerAttributes(Issue issue, IResource resource, IMarker mark
 		// Do this in one single setAttributes() call, as each set of an attribute is a workspace operation
 		Map<String, Object> attributes = new HashMap<>(16);
 
-		String lineNR = "";
 		if (issue.getLineNumber() != null) {
-			lineNR = "line: " + issue.getLineNumber() + " ";
+			attributes.put(IMarker.LINE_NUMBER, issue.getLineNumber());
+		} else {
+			attributes.put(IMarker.LOCATION, resource.getFullPath().toString());
```

Comment:
I think we can always skip the location attribute. Its only usage appear to be broken if it is present.

Reply:
Right! I pushed another commit: `IMarker.LOCATION` is not set at all.